### PR TITLE
chore: add TOC & created install.md

### PR DESCRIPTION
**Pull request by hendriknielaender**
Enhance README and Add installation documentation

- Added a Table of Contents (TOC) for better navigation.
- Moved detailed installation instructions to docs/install.md.

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+   pub fn build(b: *std.Build) void {
+       const target = b.standardTargetOptions(.{});
+       const optimize = b.standardOptimizeOption(.{});
+
+   +   const opts = .{ .target = target, .optimize = optimize };
+   +   const zbench_module = b.dependency("zbench", opts).module("zbench");
+
+       const exe = b.addExecutable(.{
+           .name = "test",
+           .root_source_file = .{ .path = "src/main.zig" },
```
Comment (FObersteiner) [inline docs/install.md:42]: you might wanna change this to the `b.path(...)` syntax ^^

File: docs/install.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+exe.addAnonymousModule("zbench", .{
+    .source_file = .{ .path = "libs/zbench/zbench.zig" },
+});
+```
+
+Now you can import like this:
+
+```zig
+const zbench = @import("zbench");
+```
```
Comment (FObersteiner) [inline docs/install.md:96]: could also add a "further reading"; for instance https://ziglang.org/learn/build-system/ and https://ziggit.dev/t/build-system-tricks/ are quite useful I think

File: README.md [hunk 3]
```diff
@@ -7,92 +7,22 @@
...
-    .source_file = .{ .path = "libs/zbench/zbench.zig" },
-});
-```
-
-Now you can import like this:
-
-```zig
-const zbench = @import("zbench");
-```
+## Content
```
Comment (FObersteiner) [inline README.md:10]: looks great!

**Review by FObersteiner — APPROVED** [review]

File: docs/install.md (hunk 2 above)
Comment (hendriknielaender) [reply docs/install.md:96]: Sure, makes sense 👍

**Pull request merged**